src/System.Transactions.Local/src/System/Transactions/Transaction.cs

```diff
@@ -319,37 +317,29 @@ public override int GetHashCode()
 
         // Don't allow equals to get the identifier
         //
-        public override bool Equals(object obj)
+        public override bool Equals(object? obj)
         {
-            Transaction transaction = obj as Transaction;
-
             // If we can't cast the object as a Transaction, it must not be equal
-            // to this, which is a Transaction.
-            if (null == transaction)
-            {
-                return false;
-            }
-
-            // Check the internal transaction object for equality.
-            return _internalTransaction.TransactionHash == transaction._internalTransaction.TransactionHash;
+            // to this, which is a Transaction. Check the internal transaction object for equality.
+            return obj is Transaction transaction &&_internalTransaction.TransactionHash == transaction._internalTransaction.TransactionHash;
```

Comment:
Unfortunately, you missed a space after `&&`
```suggestion
            return obj is Transaction transaction && _internalTransaction.TransactionHash == transaction._internalTransaction.TransactionHash;
```